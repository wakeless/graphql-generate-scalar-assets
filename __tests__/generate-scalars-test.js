const generateScalars = require('../src/generateScalars');
const parseSchema = require('../src/parseSchema');
const fs = require('fs');
const path = require('path');

let types;

beforeEach(async () => {
  const rawSchema = fs.readFileSync(path.resolve('./__tests__/fixtures/schema.graphql'), 'utf8');
  schema = await parseSchema(rawSchema, './__tests__/fixtures/schema.graphql');
  types = schema.data.__schema.types;
});


describe('Generate assets', () => {
  it('should generate enums', () => {
    expect(generateScalars(types)).toMatchSnapshot();
  });
});
