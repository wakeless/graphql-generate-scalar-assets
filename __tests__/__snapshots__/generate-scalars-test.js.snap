// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate assets should generate enums 1`] = `
"// @flow

/**
 * THIS FILE IS AUTO GENERATED, DO NOT EDIT DIRECTLY.
 */

export const Query = {};

export const BaseType = {
  parentId: { type: 'ID', nullable: false, description: null },
  clientMutationId: {
    type: 'String',
    nullable: true,
    description: 'A unique identifier for the client performing the mutation.',
  },
  content: { type: 'String', nullable: false, description: null },
  description: { type: 'String', nullable: true, description: null },
  viewUrl: { type: 'URL', nullable: true, description: 'The URL used to View' },
  customerEnum: { type: 'BaseEnum', nullable: true, description: null },
};

export const BaseInput = {
  parentId: { type: 'ID', nullable: true, description: null },
};
"
`;
